lib/optimizely/event_dispatcher.rb

```diff
@@ -39,19 +39,29 @@ def initialize(logger: nil, error_handler: nil)
     #
     # @param event - Event object
     def dispatch_event(event)
+      uri = URI.parse(event.url)
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.read_timeout = REQUEST_TIMEOUT
+      http.use_ssl = true
+
       if event.http_verb == :get
-        response = HTTParty.get(event.url, headers: event.headers, query: event.params, timeout: REQUEST_TIMEOUT)
+        request = Net::HTTP::Get.new(uri.request_uri)
 
       elsif event.http_verb == :post
-        response = HTTParty.post(event.url,
-                                 body: event.params.to_json,
-                                 headers: event.headers,
-                                 timeout: REQUEST_TIMEOUT)
+        request = Net::HTTP::Post.new(uri.request_uri)
+        request.body = event.params.to_json
+      end
+
+      # set headers
+      event.headers&.each do |key, val|
```

Comment:
I don't see headers in old implementation.

Reply:
We did pass headers to HTTPatty.post, which takes care of setting them on it's own.